### PR TITLE
Refinements to improved map canvas handlers

### PR DESCRIPTION
**Pull request by nirvn**
@domi4484 , aimed at fixing what you spotted.

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for be099900ae1ad2b48eb1346255581d9ae30b6fde: [**arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5176-44237e57f2e7bef8677e78a36de08b3d8be5039f-android-arm64.apk)
<details>
<summary>Other architectures</summary>

- [arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5176-44237e57f2e7bef8677e78a36de08b3d8be5039f-android-arm-neon.apk)
- [x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5176-44237e57f2e7bef8677e78a36de08b3d8be5039f-android-x64.apk)
- [x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5176-44237e57f2e7bef8677e78a36de08b3d8be5039f-android-x86.apk)
- [**all access arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5176-44237e57f2e7bef8677e78a36de08b3d8be5039f-android-arm64.apk)
- [all access arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5176-44237e57f2e7bef8677e78a36de08b3d8be5039f-android-arm-neon.apk)
- [all access x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5176-44237e57f2e7bef8677e78a36de08b3d8be5039f-android-x64.apk)
- [all access x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5176-44237e57f2e7bef8677e78a36de08b3d8be5039f-android-x86.apk)
</details>

**Comment by domi4484** [comment]
There are still some issues:
- after closing the attribute form I can't move the canvas anymore 
- if I take a vertical picture it becomes horizontal when editing
- buttons are still difficult to tap

File: src/qml/MapCanvas.qml [hunk 1]
```diff
@@ -408,16 +371,76 @@ Item {
...
       }
     }
   }
 
+  PinchHandler {
+      id: pinchHandler
+      enabled: interactive
+      target: null
+      acceptedDevices: PointerDevice.TouchScreen
+      grabPermissions: PointerHandler.ApprovesTakeOverByHandlersOfDifferentType
```
Comment (nirvn) [inline src/qml/MapCanvas.qml:387]:
```suggestion
      grabPermissions: PointerHandler.CanTakeOverFromHandlersOfDifferentType | PointerHandler.ApprovesTakeOverByItems
```

File: src/core/drawingcanvas.cpp [hunk 2]
```diff
@@ -43,7 +44,9 @@ void DrawingCanvas::createBlankCanvas( int width, int height, QColor backgroundC
 
 void DrawingCanvas::createCanvasFromImage( const QString &path )
 {
-  mBackgroundImage = QImage( path.startsWith( QStringLiteral( "file://" ) ) ? path.mid( 7 ) : path );
+  QImageReader imageReader( path.startsWith( QStringLiteral( "file://" ) ) ? path.mid( 7 ) : path );
```
[the remark this answers is not in the document]
Comment (domi4484) [reply src/core/drawingcanvas.cpp:47]:
```suggestion
  QImageReader imageReader( path.startsWith( QStringLiteral( "file://" ) ) ? QUrl(path).toLocalFile(): path );
```

I usually use `QUrl(path).toLocalFile()`

**Review by domi4484 — APPROVED** [review]

**Review by domi4484 — APPROVED** [review]

**Pull request merged**